bin/faucet/src/main.rs

```diff
@@ -229,3 +196,133 @@ fn long_version() -> LongVersion {
+                .layer(SetResponseHeaderLayer::if_not_present(
+                    http::header::CACHE_CONTROL,
+                    HeaderValue::from_static("no-cache"),
+                ))
+                .layer(
+                    CorsLayer::new()
+                        .allow_origin(tower_http::cors::Any)
+                        .allow_methods(tower_http::cors::Any),
+                ),
+        )
+        .with_state(faucet_state)
+}
+
+async fn serve_faucet(config: FaucetConfig) -> anyhow::Result<()> {
+    let faucet_state = FaucetState::new(config.clone()).await?;
+
+    info!(target: COMPONENT, %config, "Initializing server");
+
+    let app = create_new_router(faucet_state);
+
+    let socket_addr =
+        config
+            .endpoint
+            .socket_addrs(|| None)?
+            .into_iter()
+            .next()
+            .ok_or(anyhow::anyhow!(
+                "Couldn't get any socket addrs for endpoint: {}",
+                config.endpoint
+            ))?;
+    let listener = TcpListener::bind(socket_addr).await.context("failed to bind TCP listener")?;
+
+    info!(target: COMPONENT, endpoint = %config.endpoint, "Server started");
+
+    axum::serve(listener, app).await.unwrap();
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use std::{process::Command, str::FromStr, time::Duration};
+
+    use fantoccini::ClientBuilder;
+    use serde_json::{json, Map};
+    use tokio::time::sleep;
+    use url::Url;
+
+    use crate::{config::FaucetConfig, serve_faucet, stub_rpc_api::serve_stub};
+
+    #[tokio::test]
+    async fn test_website() {
+        let stub_node_url = Url::from_str("http://localhost:50051").unwrap();
+
+        // Start the stub node
+        let stub_url_clone = stub_node_url.clone();
+        tokio::spawn(async move { serve_stub(&stub_url_clone).await.unwrap() });
+
+        // Start the faucet connected to the stub
+        let config = FaucetConfig {
+            node_url: stub_node_url,
+            faucet_account_path: "src/test_data/faucet.mac".into(),
+            ..FaucetConfig::default()
+        };
+        let website_url = config.endpoint.clone();
+        tokio::spawn(async move { serve_faucet(config).await.unwrap() });
+
+        // Start chromedriver. This requires having chromedriver and chrome installed
+        let chromedriver_port = "57709";
+        #[allow(clippy::zombie_processes)]
+        let mut chromedriver = Command::new("chromedriver")
+            .arg(format!("--port={chromedriver_port}"))
+            .spawn()
+            .expect("failed to start chromedriver");
+
+        // Wait for the server to be running
+        sleep(Duration::from_secs(1)).await;
+
+        // Start fantoccini client
+        let mut caps = Map::new();
+        caps.insert(
+            "goog:chromeOptions".to_string(),
+            json!({"args": ["--headless", "--disable-gpu", "--no-sandbox"]}),
+        );
+        let client = ClientBuilder::native()
+            .capabilities(caps)
```

Comment:
```suggestion
        let client = ClientBuilder::native()
            .capabilities([(
                "goog:chromeOptions".to_string(),
                json!({"args": ["--headless", "--disable-gpu", "--no-sandbox"]})
            )].into()),
```
Might be able to do something like this - and do they really abbreviate it to `goog:`?